chat.go

```diff
@@ -89,18 +89,18 @@ type ChatCompletionResponseFormat struct {
 
 // ChatCompletionRequest represents a request structure for chat completion API.
 type ChatCompletionRequest struct {
-	Model            string                       `json:"model"`
-	Messages         []ChatCompletionMessage      `json:"messages"`
-	MaxTokens        int                          `json:"max_tokens,omitempty"`
-	Temperature      float32                      `json:"temperature,omitempty"`
-	TopP             float32                      `json:"top_p,omitempty"`
-	N                int                          `json:"n,omitempty"`
-	Stream           bool                         `json:"stream,omitempty"`
-	Stop             []string                     `json:"stop,omitempty"`
-	PresencePenalty  float32                      `json:"presence_penalty,omitempty"`
-	ResponseFormat   ChatCompletionResponseFormat `json:"response_format,omitempty"`
-	Seed             *int                         `json:"seed,omitempty"`
-	FrequencyPenalty float32                      `json:"frequency_penalty,omitempty"`
+	Model            string                        `json:"model"`
+	Messages         []ChatCompletionMessage       `json:"messages"`
+	MaxTokens        int                           `json:"max_tokens,omitempty"`
+	Temperature      float32                       `json:"temperature,omitempty"`
+	TopP             float32                       `json:"top_p,omitempty"`
+	N                int                           `json:"n,omitempty"`
+	Stream           bool                          `json:"stream,omitempty"`
+	Stop             []string                      `json:"stop,omitempty"`
+	PresencePenalty  float32                       `json:"presence_penalty,omitempty"`
+	ResponseFormat   *ChatCompletionResponseFormat `json:"response_format,omitempty"`
```

Comment:
Change itself LG; maybe consider adding a unit test for issue #541?

Reply:
@mikeb26 @henomis Thanks for the quick fix. It's self-evident that if the struct is nil, the field will be omitted due to 'omitempty', so I think it's unnecessary to write a unit test for that. It would be better to ensure that the integration test succeeds locally.
refs: https://github.com/sashabaranov/go-openai/blob/master/CONTRIBUTING.md#integration-test

Reply:
Oops, it seems like the integration test is currently failing to compile.

Reply:
I fixed the integration test on my local machine and tested it. The response was successful!

**Before (in master):**
```sh
$ go test -v -tags=integration ./api_integration_test.go
=== RUN   TestAPI
    api_integration_test.go:61: error, status code: 400, message: '' is not one of ['json_object', 'text'] - 'response_format.type' [CreateChatCompletion (without name) returned error]
    api_integration_test.go:76: error, status code: 400, message: '' is not one of ['json_object', 'text'] - 'response_format.type' [CreateChatCompletion (with name) returned error]
    api_integration_test.go:132: error, status code: 400, message: '' is not one of ['json_object', 'text'] - 'response_format.type' [CreateChatCompletion (with functions) returned error]
--- FAIL: TestAPI (2.76s)
=== RUN   TestAPIError
--- PASS: TestAPIError (0.19s)
FAIL
FAIL	command-line-arguments	3.230s
FAIL
```

**After(in henomis:fix/514-use-pointer-for-ChatCompletionResponseFormat):**
```sh
go test -v -tags=integration ./api_integration_test.go
=== RUN   TestAPI
--- PASS: TestAPI (5.51s)
=== RUN   TestAPIError
--- PASS: TestAPIError (0.19s)
PASS
ok  	command-line-arguments	5.991s
```

Reply:
Thanks!